lstchain/calib/camera/calibrator.py

```diff
@@ -128,9 +154,25 @@ def _calibrate_dl1(self, event, telid):
             self.image_extractor.neighbors = camera.neighbor_matrix_where
         charge, pulse_time = self.image_extractor(waveforms)
 
-        event.dl0.event_id = event.r1.event_id
-        event.dl1.tel[telid].image = charge
-        event.dl1.tel[telid].pulse_time = pulse_time + self.mon_data.tel[telid].calibration.time_correction
-
+        # correct time with drs4 correction if available
+        if self.time_corrector:
+            pulse_corr_array = self.time_corrector.get_corr_pulse(event, pulse_time)
+
+        # otherwise use the ff time correction (not drs4 corrected)
+        else:
+            pulse_corr_array = pulse_time + self.mon_data.tel[telid].calibration.time_correction
+
+        # perform the gain selection if the threshold is defined
+        if self.gain_threshold:
+            waveforms, gain_mask = self.gain_selector(event.r1.tel[telid].waveform)
+            event.dl1.tel[telid].image = charge[gain_mask, np.arange(charge.shape[1])]
+            event.dl1.tel[telid].pulse_time = pulse_corr_array[gain_mask, np.arange(pulse_corr_array.shape[1])]
+            # remember the mask in the pixel_status array (this info  is missing for the moment in the
+            # r1 cantainer)
+            event.lst.tel[telid].evt.pixel_status = gain_mask
+
+        else:
+            event.dl1.tel[telid].image = charge
+            event.dl1.tel[telid].pulse_time = pulse_corr_array
```

Comment:
In this case, the image and pulse_time will be of shape `(1855,2)` instead of `(1855,)`.
I don't think this is what we want.
Instead, I'd recommend a ManualGainSelection to select high gain by default.

Reply:
And then the notebook will need to be modified to plot images with single dimension only.

Reply:
Hi @vuillaut, 
actually this is what I want: I would like to have the possibility to calibrate without performing any gain selection (mainly for debug purposes). The HG by default can be obtained just putting the default threshold to 4095 ;-)
By the way, here I fixed the gain selector to the Threshold one, should we give the possibility to choose other selectors?

Reply:
I understand the advantage of having this for debugging but here the default option is to not select gains (since `None` is the default for `gain_threshold`). Maybe the default should be to have a gain selection to make sure that image and pulse_time have the correct dimension by default? And keeping both gains be a "hidden feature" for expert and debug?

To your question, yes I think it would be better to have the gain selector be the one passed in the config file.

Reply:
> I understand the advantage of having this for debugging but here the default option is to not select gains (since `None` is the default for `gain_threshold`). Maybe the default should be to have a gain selection to make sure that image and pulse_time have the correct dimension by default? And keeping both gains be a "hidden feature" for expert and debug?

No problem, I can make 4094 as default
> 
> To your question, yes I think it would be better to have the gain selector be the one passed in the config file.

Then I must take out the threshold trailet and pass the gain selector and one must not forget to pass the selector Config. @rlopezcoto what do you think?

Reply:
> And then the notebook will need to be modified to plot images with single dimension only.

which notebook?

Reply:
The one in this PR

Reply:
This is already working with images with single dimansion